slam-libraries/viam-orb-slam3/orbslam_server_v1.cc

```diff
@@ -778,13 +779,24 @@ double ReadTimeFromFilename(string filename) {
 
     // Now we read from buffer using get_time manipulator
     // and formatting the input appropriately.
-    ss >> std::get_time(&dt, "%Y-%m-%dT%H_%M_%SZ");
-    double sub_sec =
-        (double)std::stof(filename.substr(filename.find(".")), &sz);
+    ss >> std::get_time(&dt, time_format.c_str());
     time_t thisTime = std::mktime(&dt);
+    auto sub_sec_index = filename.find(".");
+    if ((sub_sec_index != string::npos)) {
+        double sub_sec = (double)std::stof(filename.substr(sub_sec_index), &sz);
+        double myTime = (double)thisTime + sub_sec;
+        return myTime;
+    } else {
+        return (double)thisTime;
+    }
+}
 
-    double myTime = (double)thisTime + sub_sec;
-    return myTime;
+// Checks if using new or old time format
+void FindTimeFormat(string filename) {
```

Comment:
function intended to check which format is being used. Can be removed after PR in rdk goes thru